modules/product/src/main/java/com/opengamma/strata/product/bond/ResolvedBill.java

```diff
@@ -0,0 +1,742 @@
+/*
+ * Copyright (C) 2018 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.bond;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.gen.BeanDefinition;
+import org.joda.beans.gen.PropertyDefinition;
+
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.basics.date.DaysAdjustment;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.ResolvedProduct;
+import com.opengamma.strata.product.SecurityId;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * A bill, resolved for pricing.
+ */
+@BeanDefinition(constructorScope = "package")
+public class ResolvedBill
+    implements ResolvedProduct, ImmutableBean, Serializable {
+
+  /**
+   * The security identifier.
+   * <p>
+   * This identifier uniquely identifies the security within the system.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final SecurityId securityId;
+  /**
+   * The currency of the bill notional.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final Currency currency;
+  /**
+   * The notional amount, must be positive.
+   * <p>
+   * The notional expressed here must be positive.
+   * The currency of the notional is specified by {@code currency}.
+   */
+  @PropertyDefinition(validate = "ArgChecker.notNegativeOrZero")
+  private final double notional;
```

Comment:
Maybe currency and notional should be combined to be `CurrencyAmount` ? Or combined with a date to make a `Payment` ? Depends on what the pricer needs. (Fixed Coupon Bonds combines fields like this)